specification/logs/semantic_conventions/Instrumentation/mobile-events.md

```diff
@@ -0,0 +1,50 @@
+# Semantic Conventions for mobile instrumentation
+
+**Status**: [Experimental][DocumentStatus]
+
+This document defines semantic conventions for instrumentation on mobile platforms.
+
+<!-- toc -->
+
+- [Lifecycle instrumentation](#lifecycle-instrumentation)
+  * [`state` values](#state-values)
+    + [iOS](#ios)
+    + [Android](#android)
+
+<!-- tocstop -->
+
+## Lifecycle instrumentation
+
+This section defines how to apply semantic conventions when instrumenting application lifecycle.
+
+**Event name**: `client.lifecycle`
+
+**Event domain**: `device`
+
+The following attributes are stored in the `event.data` map.
+
+| Attribute        | Type   | Description                                | Examples          | Requirement level |
+|------------------|--------|--------------------------------------------|-------------------|-------------------|
+| `state` | String | The state entered at the time of the event | platform specific | Required          |
+
+### `state` values
+
+#### iOS
+
+| Name                           | `state` Value | description                                |
+|--------------------------------|------------------------|--------------------------------------------|
+| applicationDidBecomeActive     | `active`               | The app has become "active"                |
+| applicationWillResignActive    | `inactive`             | The app is about to become "inactive".     |
+| applicationDidEnterBackground  | `background`           | The app is now in the background.          |
+| applicationWillEnterForeground | `foreground`           | The app is about to enter the foreground.  |
+| applicationWillTerminate       | `terminate`            | The app is about to terminate.             |
+
+#### Android
+
+| Name                 | `state` Value | description                                     |
+|----------------------|------------------------|-------------------------------------------------|
+| App process onCreate | `created`              | The app's process has been launched.            |
+| App process onStart  | `started`              | The app is about to be shown in the foreground. |
+| App process onResume | `resumed`              | The app is in the foreground.                   |
+| App process onPause  | `paused`               | The app is about to go into the background.     |
+| App process onStop   | `stopped`              | The app is in the background.                   |
```

Comment:
Are we sure that these apply to the "App" or is it more appropriate for the `Activity`? I think the only one that is typically App centric is `onCreate()` but threre's also `onActivityCreated()`. I think it's probably more canonical to say `Activity` for all of these. What do you think?

Reply:
> Are we sure that these apply to the "App" or is it more appropriate for the Activity? I think the only one that is typically App centric is onCreate() but threre's also onActivityCreated(). I think it's probably more canonical to say Activity for all of these. What do you think?

Hi @breedx-splk - You're right, those names tend to be more associated with Activities, though we ended up using them in order to reference the ones from [this class](https://developer.android.com/reference/androidx/lifecycle/ProcessLifecycleOwner) which is the tool we use to tell when there's a lifecycle change in an app. Probably we can use some screen-specific names later in case we'd like to report each screen's lifecycle, though at the moment our use case was to gather app-wide lifecycle information.

>  think there should also be an entry for destroyed.

Yeah, it sounds about right, although I think it's worth noting that [this tool](https://developer.android.com/reference/androidx/lifecycle/ProcessLifecycleOwner) docs read that the `destroy` event will never be dispatched from it, and while it's not entirely impossible to record when a `destroy` state happens, it's not guaranteed either, based on what they mention [here](https://developer.android.com/reference/android/app/Activity#onDestroy()) about some situations in which the OS will kill the app process without calling the `onDestroy` methods. So that's why we didn't add it at first, however, I agree that having it here is useful to represent the overall Android lifecycle, so it's fine (as long as it's not mandatory to send it 😅)